my-webxr-app/src/components/Box.tsx

```diff
@@ -1,11 +1,32 @@
-//creating a box that accepts props to be used in the canvas
-// can surround other things in this to creat things like text boxes/menus 
-export default function Box({ color, size, scale, children, ...rest }: any) {
-    return (
-      <mesh scale={scale} {...rest}>
-        <boxGeometry args={size} />
-        <meshPhongMaterial color={color} />
-        {children}
-      </mesh>
-    )
-  }
\ No newline at end of file
+// creating a box that accepts props to be used in the canvas
+// can surround other things in this to creat things like text boxes/menus
+
+import { BoxGeometryProps } from '@react-three/fiber';
+import THREE from 'three';
+
+type BoxProps = {
+  color: THREE.Color | number;
+  size: BoxGeometryProps['args'];
+  scale: [number, number, number];
+  children?: JSX.Element;
+  rest?: JSX.IntrinsicElements['mesh'];
+};
+export default function Box({
+  color, size, scale, children, ...rest
+}: BoxProps) {
+  return (
+    // Justifiable argument for prop spreading here, since there are a large number of possible
+    // mesh props
+    // eslint-disable-next-line react/jsx-props-no-spreading
```

Comment:
With our style guide, prop spreading is not permitted, but I find it to be justified in this instance since we cannot feasibly account for all the potential mesh properties.